pom.xml

```diff
@@ -1542,7 +1542,7 @@ flexible messaging model and an intuitive client API.</description>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-surefire-plugin</artifactId>
         <configuration>
-          <argLine>${testJacocoAgentArgument} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${testHeapDumpPath} -XX:+ExitOnOutOfMemoryError -Xmx1G -XX:+UseZGC
+          <argLine>${testJacocoAgentArgument} -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=${testHeapDumpPath} -XX:+ExitOnOutOfMemoryError -Xmx1532M -XX:+UseZGC
```

Comment:
Instead of changing this, make the parameter configurable so that this can be increased if someone is running all tests together. For Pulsar CI, it's better to keep this value at 1G.

So what you could do is adding a separate maven property "testMaxHeapSize" so that it can be passed from the command line with `-DtestMaxHeapSize=1532M`. Makes sense?